java-samples/corda5-negotiation-cordapp/workflows/src/main/java/com/r3/developers/samples/negotiation/workflows/accept/AcceptFlowRequest.java

```diff
@@ -0,0 +1,116 @@
+package com.r3.developers.samples.negotiation.workflows.accept;
+
+import com.r3.developers.samples.negotiation.contracts.ProposalAndTradeContract;
+import com.r3.developers.samples.negotiation.states.Member;
+import com.r3.developers.samples.negotiation.states.Proposal;
+import com.r3.developers.samples.negotiation.states.Trade;
+import com.r3.developers.samples.negotiation.workflows.modify.ModifyFlowRequest;
+import net.corda.v5.application.flows.*;
+import net.corda.v5.application.marshalling.JsonMarshallingService;
+import net.corda.v5.application.membership.MemberLookup;
+import net.corda.v5.application.messaging.FlowMessaging;
+import net.corda.v5.application.messaging.FlowSession;
+import net.corda.v5.base.annotations.Suspendable;
+import net.corda.v5.base.exceptions.CordaRuntimeException;
+import net.corda.v5.ledger.common.NotaryLookup;
+import net.corda.v5.ledger.utxo.StateAndRef;
+import net.corda.v5.ledger.utxo.UtxoLedgerService;
+import net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction;
+import net.corda.v5.ledger.utxo.transaction.UtxoTransactionBuilder;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@InitiatingFlow(protocol = "accept")
+public class AcceptFlowRequest implements ClientStartableFlow {
+
+    private final static Logger log = LoggerFactory.getLogger(AcceptFlowRequest.class);
+
+    @CordaInject
+    public FlowMessaging flowMessaging;
+
+    @CordaInject
+    public JsonMarshallingService jsonMarshallingService;
+
+    @CordaInject
+    public MemberLookup memberLookup;
+
+    @CordaInject
+    NotaryLookup notaryLookup;
+
+    @CordaInject
+    UtxoLedgerService utxoLedgerService;
+
+    @CordaInject
+    public FlowEngine flowEngine;
+
+    @Suspendable
+    @Override
+    public String call(@NotNull ClientRequestBody requestBody) {
+        AcceptFlowArgs request = requestBody.getRequestBodyAs(jsonMarshallingService,AcceptFlowArgs.class);
+        // Get UUID from input JSON
+        UUID proposalID = request.getProposalID();
+
+        UUID stateID = utxoLedgerService.findUnconsumedStatesByType(Proposal.class).get(0).getState().getContractState().getProposalID();
+
+        // Getting the old Proposal State as an input state
+        List<StateAndRef<Proposal>> proposalStatAndRef = utxoLedgerService.findUnconsumedStatesByType(Proposal.class);
+        List< StateAndRef<Proposal>> proposalStatAndRefWithId = proposalStatAndRef.stream().
+                filter(it->it.getState().getContractState().getProposalID().equals(proposalID)).collect(Collectors.toList());
+
+        if(proposalStatAndRefWithId.size()!=1) throw  new CordaRuntimeException("Multiple or zero Proposal states with id " + proposalID + "found");
```

Comment:
Are you able to extract the exception message to a field?

Reply:
sure, I'll extract all such messages into their fields.